lambdas/abandon/src/test/java/uk/gov/di/ipv/cri/kbv/api/handler/AbandonKbvHandlerTest.java

```diff
@@ -80,6 +96,39 @@ void shouldReturnErrorBadRequestWhenItReceivesARequestThatDoesNotContainASession
         verify(mockEventProbe).counterMetric(ABANDON_KBV, 0d);
     }
 
+    @Test
+    void shouldReturnErrorBadRequestWhenItCannotSendAuditEvent() throws SqsException {
+        Map<String, String> sessionHeader = Map.of(HEADER_SESSION_ID, UUID.randomUUID().toString());
+        when(input.getHeaders()).thenReturn(sessionHeader);
+
+        KBVItem kbvItem = new KBVItem();
+        when(mockKbvStorageService.getKBVItem(
+                        UUID.fromString(sessionHeader.get(HEADER_SESSION_ID))))
+                .thenReturn(kbvItem);
+        SessionItem mockSessionItem = mock(SessionItem.class);
+        when(mockSessionService.getSession(sessionHeader.get(HEADER_SESSION_ID)))
+                .thenReturn(mockSessionItem);
+        when(mockEventProbe.log(any(ERROR.getClass()), any(SqsException.class)))
+                .thenReturn(mockEventProbe);
+        when(mockEventProbe.counterMetric(ABANDON_KBV)).thenReturn(mockEventProbe);
+
+        doNothing().when(mockSessionService).createAuthorizationCode(mockSessionItem);
+        doThrow(SqsException.class).when(mockAuditService).sendAuditEvent(anyString(), any());
+
+        var result = abandonKbvHandler.handleRequest(input, mock(Context.class));
+
+        // verify(mockAuditService)
```

Comment:
Please could you remove

Reply:
change will be committed after the next sonarCloud check